src/pages/payments/create/Create.tsx

```diff
@@ -0,0 +1,364 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @copyright Copyright (c) 2021. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @copyright Copyright (c) 2021. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { Button, InputField, SelectField } from '@invoiceninja/forms';
+import axios, { AxiosError } from 'axios';
+import paymentType from 'common/constants/payment-type';
+import { InvoiceStatus } from 'common/enums/invoice-status';
+import { endpoint } from 'common/helpers';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { Client } from 'common/interfaces/client';
+import { Invoice } from 'common/interfaces/invoice';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { useClientsQuery } from 'common/queries/clients';
+import { defaultHeaders } from 'common/queries/common/headers';
+import { useBlankPaymentQuery } from 'common/queries/payments';
+import { Alert } from 'components/Alert';
+import { Container } from 'components/Container';
+import { ConvertCurrency } from 'components/ConvertCurrency';
+import { CustomField } from 'components/CustomField';
+import Toggle from 'components/forms/Toggle';
+import { Default } from 'components/layouts/Default';
+import { useFormik } from 'formik';
+import { useEffect, useState } from 'react';
+import toast from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { generatePath, useNavigate, useParams } from 'react-router-dom';
+import { useAllInvoicesQuery } from '../common/helpers/invoices-query';
+export function Create() {
+  const { client_id } = useParams();
+  const [t] = useTranslation();
+  const { data: payment } = useBlankPaymentQuery();
+  const { data: clients } = useClientsQuery();
+  const [errors, setErrors] = useState<ValidationBag>();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const company = useCurrentCompany();
+  const [invoices, setinvoices] = useState<string[]>([]);
+  const [invoicedata, setinvoicedata] = useState<Invoice[]>([]);
+  const [convertCurrency, setconvertCurrency] = useState(false);
+  const [emailInvoice, setemailInvoice] = useState(false);
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      amount: 0,
+
+      client_id: client_id || '',
+      date: payment?.data.data.date,
+      transaction_reference: '',
+      type_id: '',
+      private_notes: '',
+      currency_id: clients?.data.data.find(
+        (client: any) => client.id == client_id
+      )?.settings.currency_id,
+      exchange_rate: 1,
+      exchange_currency_id: payment?.data.data.exchange_currency_id,
+      invoices: [],
+    },
+    onSubmit: (values) => {
```

Comment:
To clean up this code, this can be extracted to the custom hook. I will do this and please follow this in future :)